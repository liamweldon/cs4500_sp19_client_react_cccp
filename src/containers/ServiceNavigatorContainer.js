import React from "react";
import ServiceNavigator from "../components/ServiceNavigator/ServiceNavigator";
import "../components/table.scss";

class ServiceNavigatorContainer extends React.Component {
  constructor(props) {
    super(props);
    this.serviceCategoryService = this.props.serviceCategoryService;
    this.state = {
      categories: []
    };
  }

  componentDidMount() {
    this.findAllServiceCategories();
  }

  findAllServiceCategories = () =>
    this.serviceCategoryService.findAllServiceCategories().then(categories => {
      this.setState({
        categories: categories
      });
    }
  );

  render() {
    return <ServiceNavigator categories={this.state.categories}/>
  }
}

export default ServiceNavigatorContainer;