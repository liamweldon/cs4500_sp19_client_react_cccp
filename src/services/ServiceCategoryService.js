export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
