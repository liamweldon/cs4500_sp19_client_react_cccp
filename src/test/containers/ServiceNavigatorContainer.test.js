import React from 'react'
import TestRenderer from "react-test-renderer";

import ServiceNavigatorContainer from '../../containers/ServiceNavigatorContainer'
import mockService from  '../services/mocks/MockServiceCategoryService'
import ServiceCategoryService from '../../services/ServiceCategoryService'
import serviceCategories from '../mockData/service-categories.mock.json'

// Service mocking code
const serviceCategoryService = ServiceCategoryService.getInstance();

// Mock the service here
beforeAll(() => {
  global.fetch = mockService(serviceCategories);
});

test('ServiceNavigatorContainer renders correctly', () => {
    const testRenderer = TestRenderer.create(
        <ServiceNavigatorContainer
            serviceCategoryService={serviceCategoryService}/>)
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance   = testRenderer.root
    let   searchProviderNames = testInstance.findAllByProps({className: 'form-control'})

    searchProviderNames[0].props.onChange({target: {value: 'J123'}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})
