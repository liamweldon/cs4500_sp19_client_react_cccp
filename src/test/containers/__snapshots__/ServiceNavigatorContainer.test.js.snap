// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceNavigatorContainer renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <div>
        <div>
          <div
            className="input-group input-group-lg"
          >
            <input
              className="form-control"
              name="providerSearchTerm"
              onChange={[Function]}
              placeholder="Search for providers"
              type="text"
              value=""
            />
            <input
              className="form-control"
              name="zipSearchTerm"
              onChange={[Function]}
              placeholder="Zip code"
              type="text"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <br />
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <div
        style={
          Object {
            "position": "fixed",
          }
        }
      >
        <ul
          className="list-group"
        />
      </div>
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;

exports[`ServiceNavigatorContainer renders correctly 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <div>
        <div>
          <div
            className="input-group input-group-lg"
          >
            <input
              className="form-control"
              name="providerSearchTerm"
              onChange={[Function]}
              placeholder="Search for providers"
              type="text"
              value="J123"
            />
            <input
              className="form-control"
              name="zipSearchTerm"
              onChange={[Function]}
              placeholder="Zip code"
              type="text"
              value=""
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Search
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-3 text-right"
    >
      <a
        href="#"
      >
        Sign up
      </a>
    </div>
    <div
      className="col-1"
    >
      <a
        href="#"
      >
        Log in
      </a>
    </div>
  </div>
  <br />
  <br />
  <div
    className="row"
  >
    <div
      className="col-3"
    >
      <div
        style={
          Object {
            "position": "fixed",
          }
        }
      >
        <ul
          className="list-group"
        />
      </div>
    </div>
    <div
      className="col-9"
    >
      <ul
        className="list-group no-border"
      />
    </div>
  </div>
</div>
`;
